src/modules/esl-popup/core/esl-popup-position.ts

```diff
@@ -148,72 +158,75 @@ function adjustAlongMajorAxis(cfg: PopupPositionConfig, value: PopupPositionValu
-     *
-     * const func = isStartingSide ? Math.max : Math.min;
-     * arrowAdjust = func(diffCoord, arrowLimit - arrowCoord);
-     */
   }
 
   return arrowAdjust;
 }
 
+/**
+ * Sets up the configuration for adjusting position along the minor axis
+ * @param cfg - popup position config
+ * @param popup - current popup's position value
+ * @returns configuration for adjusting position along the minor axis
+ */
+function setupAlignmentBySide(cfg: PopupPositionConfig, popup: Rect): PopupAlignmentConfig {
+  const isHorizontal = isOnHorizontalAxis(cfg.position);
+  const start = isHorizontal ? 'y' : 'x';
+  const end = isHorizontal ? 'bottom' : 'right';
+  const dimension = isHorizontal ? 'height' : 'width';
+  const isOutAtStart = popup[start] < cfg.outer[start];
+  const isOutAtEnd = popup[end] > cfg.outer[end];
+  const isWider = cfg.outer[dimension] < cfg.element[dimension];
+
+  return {isHorizontal, start, end, isOutAtStart, isOutAtEnd, isWider};
+}
+
 /**
  * Updates popup and arrow positions to fit on minor axis.
  * @param cfg - popup position config
  * @param value - current popup's position value
  * @returns updated popup position value
  * */
 function fitOnMinorAxis(cfg: PopupPositionConfig, value: PopupPositionValue): PopupPositionValue {
-  if (cfg.behavior !== 'fit' && cfg.behavior !== 'fit-minor') return value;
+  if (!['fit', 'fit-minor'].includes(cfg.behavior)) return value;
 
-  const isHorizontal = isOnHorizontalAxis(cfg.position);
-  const start = isHorizontal ? 'y' : 'x';
-  const end = isHorizontal ? 'bottom' : 'right';
-  const dimension = isHorizontal ? 'height' : 'width';
+  const {popup, arrow} = value;
+  const {isHorizontal, start, end, isOutAtStart, isOutAtEnd, isWider} = setupAlignmentBySide(cfg, popup);
 
-  if (cfg.outer[dimension] < cfg.element[dimension] ||  // cancel fit mode if the popup size is greater than the outer limiter size
-      cfg.trigger[start] < cfg.outer[start] ||          // or the trigger is outside the outer limiting element
-      cfg.trigger[end] > cfg.outer[end]
-  ) return value;
+  // nothing to do when there is no outing
+  if (!isOutAtStart && !isOutAtEnd) return value;
+  // start-side adjusting happens if there is only start-side outing or LTR content direction
+  const isStarting = isOutAtStart && (!isOutAtEnd || !cfg.isRTL);
 
-  const {popup, arrow} = value;
-  // check the starting side of the axis
-  let arrowLimit = cfg.marginArrow;
-  let coordAdjust = adjustAlignmentBySide(popup[start] - cfg.outer[start], arrow[start], arrowLimit, true);
-  if (coordAdjust) {
-    popup[start] -= coordAdjust;
-    arrow[start] += coordAdjust;
+  let diff = isStarting ? popup[start] - cfg.outer[start] : popup[end] - cfg.outer[end];
+  if (isWider) {
+    // apply another value on case when popup is greater than the outer container
+    diff = cfg.isRTL ? popup[end] - cfg.outer[end] : popup[start] - cfg.outer[start];
   }
```

Comment:
One more cosmetic (but immutable) idea
```suggestion
  const diffSide = (isWider ? !cfg.isRTL : isStarting) ? start : end;
  const diff = popup[diffSide] - cfg.outer[diffSide];
```

Reply:
suggestion was applied with explanation comment